### ISSUE TITLE
Called process exception in felix logs running ip6tables

### ISSUE THREAD

**Issue by robbrockbank** [Metaswitch/calico#368]
Spotted this in the felix.logs - I'm afraid I can't indicate what was happening at the time or if there were any external symptoms.

I can provide the full log on request.

```
2015-04-15 09:22:23,769 [INFO][3] calico.felix.fetcd 110: (Re)connecting to etcd...
2015-04-15 09:22:23,769 [INFO][3] calico.felix.fetcd 87: Waiting for etcd to be ready and for config to be present.
2015-04-15 09:22:23,770 [INFO][3] urllib3.connectionpool 258: Starting new HTTP connection (1): localhost
2015-04-15 09:22:23,772 [INFO][3] calico.felix.fetcd 102: etcd is ready.
2015-04-15 09:22:23,773 [INFO][3] calico.felix.splitter 46: Applying snapshot. STAGE 1a: rules.
2015-04-15 09:22:23,774 [INFO][3] calico.felix.splitter 49: Applying snapshot. STAGE 1b: tags.
2015-04-15 09:22:23,774 [INFO][3] calico.felix.splitter 56: Applying snapshot. STAGE 2: endpoints->endpoint mgr.
2015-04-15 09:22:23,775 [INFO][3] calico.felix.splitter 62: Applying snapshot. DONE. 0 rules, 0 tags, 0 endpoints
2015-04-15 09:22:23,775 [INFO][3] calico.felix.fetcd 184: Initial etcd index: 68; modifiedIndex: 3
2015-04-15 09:22:23,776 [INFO][7] calico.felix.profilerules 62: Rules manager applying snapshot; 0 rules
2015-04-15 09:22:23,776 [INFO][8] calico.felix.profilerules 62: Rules manager applying snapshot; 0 rules
2015-04-15 09:22:23,878 [INFO][9] calico.felix.dispatch 58: Applying dispatch chains snapshot.
2015-04-15 09:22:23,878 [INFO][9] calico.felix.dispatch 109: DispatchChains<ipv4,entries=0> Updating dispatch chain, num entries: 0
2015-04-15 09:22:23,879 [INFO][10] calico.felix.dispatch 58: Applying dispatch chains snapshot.
2015-04-15 09:22:23,879 [INFO][10] calico.felix.dispatch 109: DispatchChains<ipv6,entries=0> Updating dispatch chain, num entries: 0
2015-04-15 09:22:32,309 [WARNING][3] calico.felix.fetcd 277: Global config changed but we don't yet support dynamic config key=/calico/config/InterfacePrefix
2015-04-15 09:22:32,310 [INFO][3] calico.felix.fetcd 110: (Re)connecting to etcd...
2015-04-15 09:22:32,310 [INFO][3] calico.felix.fetcd 87: Waiting for etcd to be ready and for config to be present.
2015-04-15 09:22:32,311 [INFO][3] urllib3.connectionpool 258: Starting new HTTP connection (1): localhost
2015-04-15 09:22:32,314 [INFO][3] calico.felix.fetcd 102: etcd is ready.
2015-04-15 09:22:32,315 [INFO][3] calico.felix.splitter 46: Applying snapshot. STAGE 1a: rules.
2015-04-15 09:22:32,316 [INFO][3] calico.felix.splitter 49: Applying snapshot. STAGE 1b: tags.
2015-04-15 09:22:32,317 [INFO][3] calico.felix.splitter 56: Applying snapshot. STAGE 2: endpoints->endpoint mgr.
2015-04-15 09:22:32,317 [INFO][3] calico.felix.splitter 62: Applying snapshot. DONE. 0 rules, 0 tags, 0 endpoints
2015-04-15 09:22:32,317 [INFO][3] calico.felix.fetcd 184: Initial etcd index: 69; modifiedIndex: 3
2015-04-15 09:22:32,318 [INFO][7] calico.felix.profilerules 62: Rules manager applying snapshot; 0 rules
2015-04-15 09:22:32,318 [INFO][8] calico.felix.profilerules 62: Rules manager applying snapshot; 0 rules
2015-04-15 09:22:32,334 [WARNING][3] calico.felix.fetcd 277: Global config changed but we don't yet support dynamic config key=/calico/config/Ready
2015-04-15 09:22:32,334 [INFO][3] calico.felix.fetcd 110: (Re)connecting to etcd...
2015-04-15 09:22:32,335 [INFO][3] calico.felix.fetcd 87: Waiting for etcd to be ready and for config to be present.
2015-04-15 09:22:32,335 [INFO][3] urllib3.connectionpool 258: Starting new HTTP connection (1): localhost
2015-04-15 09:22:32,339 [INFO][3] calico.felix.fetcd 102: etcd is ready.
2015-04-15 09:22:32,341 [INFO][3] calico.felix.splitter 46: Applying snapshot. STAGE 1a: rules.
2015-04-15 09:22:32,341 [INFO][3] calico.felix.splitter 49: Applying snapshot. STAGE 1b: tags.
2015-04-15 09:22:32,342 [INFO][3] calico.felix.splitter 56: Applying snapshot. STAGE 2: endpoints->endpoint mgr.
2015-04-15 09:22:32,343 [INFO][3] calico.felix.splitter 62: Applying snapshot. DONE. 0 rules, 0 tags, 0 endpoints
2015-04-15 09:22:32,343 [INFO][3] calico.felix.fetcd 184: Initial etcd index: 70; modifiedIndex: 3
2015-04-15 09:22:32,344 [INFO][7] calico.felix.profilerules 62: Rules manager applying snapshot; 0 rules
2015-04-15 09:22:32,345 [INFO][8] calico.felix.profilerules 62: Rules manager applying snapshot; 0 rules
2015-04-15 09:22:32,420 [INFO][9] calico.felix.dispatch 58: Applying dispatch chains snapshot.
2015-04-15 09:22:32,420 [INFO][9] calico.felix.dispatch 58: Applying dispatch chains snapshot.
2015-04-15 09:22:32,421 [INFO][9] calico.felix.dispatch 109: DispatchChains<ipv4,entries=0> Updating dispatch chain, num entries: 0
2015-04-15 09:22:32,421 [INFO][10] calico.felix.dispatch 58: Applying dispatch chains snapshot.
2015-04-15 09:22:32,421 [INFO][10] calico.felix.dispatch 58: Applying dispatch chains snapshot.
2015-04-15 09:22:32,422 [INFO][10] calico.felix.dispatch 109: DispatchChains<ipv6,entries=0> Updating dispatch chain, num entries: 0
2015-04-15 09:22:53,785 [INFO][11] calico.felix.splitter 77: Triggering a cleanup of orphaned ipsets/chains
2015-04-15 09:22:53,786 [INFO][5] calico.felix.fiptables 218: Cleaning up left-over iptables state.
2015-04-15 09:22:53,804 [INFO][5] calico.felix.fiptables 237: Cleanup finished, deleted 0 chains, failed to delete these chains: []
2015-04-15 09:22:53,805 [INFO][6] calico.felix.fiptables 218: Cleaning up left-over iptables state.
2015-04-15 09:22:53,809 [INFO][6] calico.felix.fiptables 237: Cleanup finished, deleted 0 chains, failed to delete these chains: []
2015-04-15 09:23:02,325 [INFO][12] calico.felix.splitter 77: Triggering a cleanup of orphaned ipsets/chains
2015-04-15 09:23:02,326 [INFO][5] calico.felix.fiptables 218: Cleaning up left-over iptables state.
2015-04-15 09:23:02,335 [INFO][5] calico.felix.fiptables 237: Cleanup finished, deleted 0 chains, failed to delete these chains: []
2015-04-15 09:23:02,337 [INFO][6] calico.felix.fiptables 218: Cleaning up left-over iptables state.
2015-04-15 09:23:02,342 [INFO][6] calico.felix.fiptables 237: Cleanup finished, deleted 0 chains, failed to delete these chains: []
2015-04-15 09:23:02,345 [INFO][13] calico.felix.splitter 77: Triggering a cleanup of orphaned ipsets/chains
2015-04-15 09:23:02,347 [INFO][5] calico.felix.fiptables 218: Cleaning up left-over iptables state.
2015-04-15 09:23:02,357 [INFO][5] calico.felix.fiptables 237: Cleanup finished, deleted 0 chains, failed to delete these chains: []
2015-04-15 09:23:02,359 [INFO][6] calico.felix.fiptables 218: Cleaning up left-over iptables state.
2015-04-15 09:23:02,364 [INFO][6] calico.felix.fiptables 237: Cleanup finished, deleted 0 chains, failed to delete these chains: []
2015-04-15 09:23:02,374 [WARNING][3] calico.felix.fetcd 277: Global config changed but we don't yet support dynamic config key=/calico/config/InterfacePrefix
2015-04-15 09:23:02,375 [INFO][3] calico.felix.fetcd 110: (Re)connecting to etcd...
2015-04-15 09:23:02,375 [INFO][3] calico.felix.fetcd 87: Waiting for etcd to be ready and for config to be present.
2015-04-15 09:23:02,376 [INFO][3] urllib3.connectionpool 258: Starting new HTTP connection (1): localhost
2015-04-15 09:23:02,379 [INFO][3] calico.felix.fetcd 102: etcd is ready.
2015-04-15 09:23:02,381 [INFO][3] calico.felix.splitter 46: Applying snapshot. STAGE 1a: rules.
2015-04-15 09:23:02,381 [INFO][3] calico.felix.splitter 49: Applying snapshot. STAGE 1b: tags.
2015-04-15 09:23:02,382 [INFO][3] calico.felix.splitter 56: Applying snapshot. STAGE 2: endpoints->endpoint mgr.
2015-04-15 09:23:02,382 [INFO][3] calico.felix.splitter 62: Applying snapshot. DONE. 0 rules, 0 tags, 0 endpoints
2015-04-15 09:23:02,382 [INFO][3] calico.felix.fetcd 184: Initial etcd index: 71; modifiedIndex: 3
2015-04-15 09:23:02,383 [INFO][7] calico.felix.profilerules 62: Rules manager applying snapshot; 0 rules
2015-04-15 09:23:02,383 [INFO][8] calico.felix.profilerules 62: Rules manager applying snapshot; 0 rules
2015-04-15 09:23:02,397 [WARNING][3] calico.felix.fetcd 277: Global config changed but we don't yet support dynamic config key=/calico/config/Ready
2015-04-15 09:23:02,397 [INFO][3] calico.felix.fetcd 110: (Re)connecting to etcd...
2015-04-15 09:23:02,397 [INFO][3] calico.felix.fetcd 87: Waiting for etcd to be ready and for config to be present.
2015-04-15 09:23:02,398 [INFO][3] urllib3.connectionpool 258: Starting new HTTP connection (1): localhost
2015-04-15 09:23:02,400 [INFO][3] calico.felix.fetcd 102: etcd is ready.
2015-04-15 09:23:02,402 [INFO][3] calico.felix.splitter 46: Applying snapshot. STAGE 1a: rules.
2015-04-15 09:23:02,402 [INFO][3] calico.felix.splitter 49: Applying snapshot. STAGE 1b: tags.
2015-04-15 09:23:02,403 [INFO][3] calico.felix.splitter 56: Applying snapshot. STAGE 2: endpoints->endpoint mgr.
2015-04-15 09:23:02,403 [INFO][3] calico.felix.splitter 62: Applying snapshot. DONE. 0 rules, 0 tags, 0 endpoints
2015-04-15 09:23:02,403 [INFO][3] calico.felix.fetcd 184: Initial etcd index: 72; modifiedIndex: 3
2015-04-15 09:23:02,404 [INFO][7] calico.felix.profilerules 62: Rules manager applying snapshot; 0 rules
2015-04-15 09:23:02,404 [INFO][8] calico.felix.profilerules 62: Rules manager applying snapshot; 0 rules
2015-04-15 09:23:02,484 [INFO][9] calico.felix.dispatch 58: Applying dispatch chains snapshot.
2015-04-15 09:23:02,484 [INFO][9] calico.felix.dispatch 58: Applying dispatch chains snapshot.
2015-04-15 09:23:02,484 [INFO][9] calico.felix.dispatch 109: DispatchChains<ipv4,entries=0> Updating dispatch chain, num entries: 0
2015-04-15 09:23:02,485 [INFO][10] calico.felix.dispatch 58: Applying dispatch chains snapshot.
2015-04-15 09:23:02,485 [INFO][10] calico.felix.dispatch 58: Applying dispatch chains snapshot.
2015-04-15 09:23:02,485 [INFO][10] calico.felix.dispatch 109: DispatchChains<ipv6,entries=0> Updating dispatch chain, num entries: 0
2015-04-15 09:23:32,411 [INFO][14] calico.felix.splitter 77: Triggering a cleanup of orphaned ipsets/chains
2015-04-15 09:23:32,413 [INFO][15] calico.felix.splitter 77: Triggering a cleanup of orphaned ipsets/chains
2015-04-15 09:23:32,414 [INFO][5] calico.felix.fiptables 218: Cleaning up left-over iptables state.
2015-04-15 09:23:32,423 [INFO][5] calico.felix.fiptables 237: Cleanup finished, deleted 0 chains, failed to delete these chains: []
2015-04-15 09:23:32,424 [INFO][5] calico.felix.fiptables 218: Cleaning up left-over iptables state.
2015-04-15 09:23:32,427 [INFO][6] calico.felix.fiptables 218: Cleaning up left-over iptables state.
2015-04-15 09:23:32,432 [ERROR][6] calico.felix.actor 235: Exception processing c4b17c95a2b7 (cleanup)
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/calico/felix/actor.py", line 233, in _step
    result = msg.method()
  File "/usr/lib/python2.7/dist-packages/calico/felix/fiptables.py", line 225, in cleanup
    unreferenced_chains = self._load_unreferenced_chains()
  File "/usr/lib/python2.7/dist-packages/calico/felix/fiptables.py", line 142, in _load_unreferenced_chains
    "--table", self.table])
  File "/usr/lib/python2.7/dist-packages/gevent/subprocess.py", line 167, in check_output
    raise ex
CalledProcessError: Command '['ip6tables', '--list', '--table', 'filter']' returned non-zero exit status 4
2015-04-15 09:23:32,434 [ERROR][14] calico.felix.splitter 84: iptables cleanup failed, will retry on resync.
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/calico/felix/splitter.py", line 82, in trigger_cleanup
    ipt_updater.cleanup(async=False)
  File "/usr/lib/python2.7/dist-packages/calico/felix/actor.py", line 527, in queue_fn
    return result.get()
  File "/usr/lib/python2.7/dist-packages/calico/felix/actor.py", line 474, in get
    timeout=timeout)
  File "/usr/lib/python2.7/dist-packages/gevent/event.py", line 235, in get
    raise self._exception
CalledProcessError: Command '['ip6tables', '--list', '--table', 'filter']' returned non-zero exit status 4
2015-04-15 09:23:32,437 [INFO][5] calico.felix.fiptables 237: Cleanup finished, deleted 0 chains, failed to delete these chains: []
2015-04-15 09:23:32,439 [INFO][6] calico.felix.fiptables 218: Cleaning up left-over iptables state.
2015-04-15 09:23:32,445 [INFO][6] calico.felix.fiptables 237: Cleanup finished, deleted 0 chains, failed to delete these chains: []
2015-04-15 09:23:32,511 [WARNING][3] calico.felix.fetcd 277: Global config changed but we don't yet support dynamic config key=/calico/config/InterfacePrefix
```

**Comment by fasaxc** [comment]
I wonder if we're missing a "--wait" on that call; it might just be hitting a concurrent use of iptables.  Need to check what RC 4 means in the code, it's not documented.  (And probably enhance the error reporting to show any output from iptables if it's not already.)

This only impacts start-of-day cleanup.

**Comment by fasaxc** [comment]
RC 4 means "failed to get the xtables lock", I think we should be passing "--wait" on this call.

**Comment by fasaxc** [comment]
Fixed in #369
